soc/xtensa/intel_adsp/tools/cavstool_client.py

```diff
@@ -36,32 +45,52 @@ def send_cmd(self, cmd):
             self.sock.connect((self.host, self.port))
             self.sock.sendall(cmd.encode("utf-8"))
             log.info(f"Sent:     {cmd}")
-            ack = str(self.sock.recv(BUF_SIZE), "utf-8")
+            ack = str(self.sock.recv(CMD_SZ), "utf-8")
```

Comment:
This does not make a lot of difference because commands have a variable and unknown size. So you could still be reading too much; still more or less the same bug. In this particular case I think it won't happen in practice because I believe you have only one RPC in flight at a time and all commands fit in a single Ethernet packet but it's still brittle.

To be fixed when when this gets "upgraded" away from bare sockets, reliability is massively better with this PR already.

Reply:
Agree with you, let's put it in the next update. Thank you!